dali/test/python/decoder/test_imgcodec.py

```diff
@@ -0,0 +1,420 @@
+        eps = 4
+    if (file_fmt == 'jpeg2k' or file_fmt == 'mixed') and img_out_type == types.YCbCr:
+        eps = 6
+    files = get_img_files(os.path.join(test_data_root, path), subdir=subdir, ext=ext)
+    compare_pipelines(
+        img_decoder_pipe("cpu", out_type=img_out_type, files=files),
+        img_decoder_pipe("mixed", out_type=img_out_type, files=files),
+        batch_size=batch_size_test, N_iterations=3, eps=eps)
+
+
+def test_image_decoder_consistency():
+    for out_img_type in [types.RGB, types.BGR, types.YCbCr, types.GRAY, types.ANY_DATA]:
+        for file_fmt in test_good_path:
+            if (file_fmt == 'jpeg' or file_fmt == 'mixed') and out_img_type != types.RGB:
+                # TODO(staniewzki) Implement color conversion in nvJPEG
+                continue
+            if (file_fmt == 'jpeg2k' or file_fmt == 'mixed') and out_img_type == types.ANY_DATA:
+                # TODO(staniewzki) Fix ANY_DATA support in JPEG2K
+                continue
+            path = os.path.join(good_path, file_fmt)
+            yield _testimpl_image_decoder_consistency, out_img_type, file_fmt, path
+
+        for file_fmt, path, ext in [("tiff", "db/single/multichannel/tiff_multichannel", 'tif'),
+                                    ("jpeg2k", "db/single/multichannel/with_alpha", 'jp2'),
+                                    ("png", "db/single/multichannel/with_alpha", 'png')]:
+            subdir = None  # In those paths the images are not organized in subdirs
+            yield _testimpl_image_decoder_consistency, out_img_type, file_fmt, path, subdir, ext
+
+
+def _testimpl_image_decoder_tiff_with_alpha_16bit(device, out_type, path, ext):
+    @pipeline_def(batch_size=1, device_id=0, num_threads=1)
+    def pipe(device, out_type, files):
+        encoded, _ = fn.readers.file(files=files)
+        decoded = fn.experimental.decoders.image(encoded, device=device, output_type=out_type)
+        peeked_shape = fn.peek_image_shape(encoded)
+        return decoded, peeked_shape
+
+    files = get_img_files(os.path.join(test_data_root, path), ext=ext, subdir=None)
+    pipe = pipe(device, out_type=out_type, files=files)
+    pipe.build()
+    out, shape = pipe.run()
+    if device == 'mixed':
+        out = out.as_cpu()
+    out = np.array(out[0])
+    shape = np.array(shape[0])
+    expected_channels = 4 if out_type == types.ANY_DATA else 1 if out_type == types.GRAY else 3
+    assert out.shape[2] == expected_channels, \
+        f"Expected {expected_channels} but got {out.shape[2]}"
+
+
+def test_image_decoder_tiff_with_alpha_16bit():
+    for device in ['cpu', 'mixed']:
+        for out_type in [types.RGB, types.BGR, types.YCbCr, types.ANY_DATA]:
+            path = "db/single/multichannel/with_alpha_16bit"
+            for ext in [("png", "tiff", "jp2")]:
+                yield _testimpl_image_decoder_tiff_with_alpha_16bit, device, out_type, path, ext
+
+
+# @pipeline_def(batch_size=batch_size_test, device_id=0, num_threads=4)
```

Comment:
I understand that the commented out tests are tests that don't apply to the new op (e.g. deprecated alias tests). I'd remove those.
If there are tests that are not yet supported but should be at some point, you can keep those commented out

Reply:
Removed